spec/spec_helper.rb

```diff
@@ -99,25 +99,46 @@
     :record => :new_episodes,  # :once is default
   }
   c.configure_rspec_metadata!
-  c.filter_sensitive_data('Settings.CAP.TOKEN_USER:Settings.CAP.TOKEN_PASS@Settings.CAP.TOKEN_URI') do
-    "#{Settings.CAP.TOKEN_USER}:#{Settings.CAP.TOKEN_PASS}@#{Settings.CAP.TOKEN_URI}"
-  end
-  c.filter_sensitive_data('private_access_token') do |interaction|
-    if interaction.request.body == "grant_type=client_credentials"
-      regex = %r("access_token":"(.*?)")
-      matches = regex.match(interaction.response.body)
-      matches.captures.first if matches.present?
-    end
-  end
   c.filter_sensitive_data('private_bearer_token') do |interaction|
     auth = interaction.request.headers['Authorization']
-    if auth.kind_of? Array
-      bearer = auth.select {|a| a =~ /bearer/i }.first
+    if auth.is_a? Array
+      bearer = auth.select { |a| a =~ /bearer/i }.first
       bearer.gsub(/bearer /i, '') if bearer
     end
   end
   c.filter_sensitive_data('Settings.SCIENCEWIRE.HOST') { Settings.SCIENCEWIRE.HOST }
   c.filter_sensitive_data('Settings.SCIENCEWIRE.LICENSE_ID') { Settings.SCIENCEWIRE.LICENSE_ID }
+
+  # CAP API filters
+  c.filter_sensitive_data('Settings.CAP.TOKEN_URI') { Settings.CAP.TOKEN_URI }
+  c.filter_sensitive_data('Settings.CAP.AUTHORSHIP_API_URI') { Settings.CAP.AUTHORSHIP_API_URI }
+  c.filter_sensitive_data('Settings.CAP.TOKEN_USER:Settings.CAP.TOKEN_PASS') do
+    Base64.strict_encode64("#{Settings.CAP.TOKEN_USER}:#{Settings.CAP.TOKEN_PASS}")
+  end
+  c.filter_sensitive_data('private_access_token') do |interaction|
+    if interaction.request.uri.include? Settings.CAP.TOKEN_PATH
+      token_match = interaction.response.body.match(/"access_token":"(.*?)"/)
+      token_match.captures.first if token_match
+    end
+  end
+  c.filter_sensitive_data('CAP-LicenseID') do |interaction|
+    if interaction.request.uri.include? Settings.CAP.AUTHORSHIP_API_PATH
+      id_match = interaction.response.body.match(/"californiaPhysicianLicense".*?:.*?"(.*?)"/)
```

Comment:
why do we need to filter `californiaPhysicianLicense`?  I thought that was a publicly available identifier.

Reply:
this filters it out of vcr_cassettes?  I don't think we necessarily need to follow the rule of if it can be public, we should store in our repo (seeing as it takes little effort to remove).   I can see some physicians (or anyone really) wanting to limit the number of places their personal information is stored (even public personal info) and so no need to add to that list of places for them